### PR TITLE
Add link to Matplotlib paper on citing page

### PR DESCRIPTION
**Pull request by dstansby**
Makes it a bit easier for readers to find the paper.

### CODE REVIEW

File: doc/_templates/citing.html [hunk 1]
```diff
@@ -5,8 +5,9 @@
 <h1>Citing Matplotlib</h1>
 <p>
 If Matplotlib contributes to a project that leads to a scientific publication,
-please acknowledge this fact by citing the project. You can use this
-BibTeX entry:
+please acknowledge this fact by citing the project. You can cite
+<a href ='http://ieeexplore.ieee.org/document/4160265/'>this paper</a> or
```
Comment (anntzer) [inline doc/_templates/citing.html]: ... by using (as they're referring to the same object)

Comment (jklymak) [reply doc/_templates/citing.html]: Strongly suggest linking by DOI so doi.org gets to take care of any changes to ieee’s links...  http://doi.org/10.1109/MCSE.2007.55

Comment (dstansby) [reply doc/_templates/citing.html]: I think the English is fine as is, if anyone wants to change feel free to force push. Have changed to the doi.org link.

**Comment by jklymak** [comment]
Sorry, couldn't help myself...  Feel free to roll back...

**Comment by anntzer** [comment]
thanks

**Comment by tacaswell** [comment]
backported to v2.1.0-doc as 713ab25315f1cdadc5dd3fae355ba186fbd5ae61

**Pull request merged**